cpp/src/community/spectral_clustering.cu

```diff
@@ -21,24 +21,20 @@
  * ---------------------------------------------------------------------------**/
 
 #include <algorithms.hpp>
-#include <graph.hpp>
 
-#include <nvgraph/include/sm_utils.h>
 #include <rmm/thrust_rmm_allocator.h>
 #include <thrust/transform.h>
 #include <ctime>
-#include <nvgraph/include/nvgraph_error.hxx>
-#include <utilities/error.hpp>
 
-#include <nvgraph/include/modularity_maximization.hxx>
-#include <nvgraph/include/nvgraph_cublas.hxx>
-#include <nvgraph/include/nvgraph_cusparse.hxx>
-#include <nvgraph/include/partition.hxx>
+#include <graph.hpp>
```

Comment:
Which `graph.hpp` is this? I noticed it was added in https://github.com/rapidsai/raft/pull/12 but it is still in cuGraph as well.
If we plan to get it from RAFT we need to figure out how to deploy it since it is a public API header of cuGraph.

Reply:
This is one from `cugraph`. I am working on a solution to remove the one in RAFT, as the SC routines should not, after more careful thought, depend on `GraphCSRView<>` (they only use the CSR data from `GraphCSRView`, anyway). 

If, at a later date we find it worth to expose the `graph.hpp` to RAFT as well, we may do it separately. But SC shouldn't need it. I think RAFT should stay graph-agnostic as much as possible.

Reply:
Agreed. I also like the solution you are proposing because it allows us to keep `graph.hpp`. This file had over 10 commits in the past month, all driven by cuGraph, so it is best if it stays with us for now.

Reply:
Done. (Already fixed in https://github.com/rapidsai/raft/pull/12 and in this PR; `cuml` side to follow soon)